specs/agents/tracing-spans-compress.md

```diff
@@ -0,0 +1,199 @@
+## Compressing spans
+
+To mitigate the potential flood of spans to a backend,
+agents SHOULD implement the strategies laid out in this section to avoid sending almost identical and very similar spans.
+
+While compressing multiple similar spans into a single composite span can't fully get rid of all the collection overhead,
+it can significantly reduce the impact on the following areas,
+with very little loss of information.
+- Agent reporter queue utilization
+- Capturing stack traces, serialization, compression, and sending events to APM Server
+- Potential to re-use span objects, significantly reducing allocations
+- Downstream effects like reducing impact on APM Server, ES storage, and UI performance
+
+
+### Consecutive-Exact-Match compression strategy
+
+One of the biggest sources of excessive data collection are n+1 type queries and repetitive requests to a cache server.
+This strategy detects consecutive spans that hold the same information (except for the duration)
+and creates a single [composite span](tracing-spans-compress.md#composite-span).
+
+```
+[                              ]
+GET /users
+ [] [] [] [] [] [] [] [] [] []
+ 10x SELECT FROM users
+```
+
+Two spans are considered to be an exact match if they are of the same kind and if their span names are equal.
```

Comment:
can we specify this more concretely by explicitly stating the fields that need to be the same.